platform/util/platform-util/src/main/java/org/codice/ddf/platform/util/properties/PropertiesLoader.java

```diff
@@ -108,108 +126,140 @@ public Properties loadProperties(String propertiesFile) {
-      Properties filtered = new Properties();
-      for (Map.Entry<?, ?> entry : properties.entrySet()) {
-        filtered.put(
-            StrSubstitutor.replaceSystemProperties(entry.getKey()),
-            StrSubstitutor.replaceSystemProperties(entry.getValue()));
+  /** Try loading properties using Spring and a provided class loader. */
+  @VisibleForTesting
+  static Properties attemptLoadWithSpringAndClassLoader(
+      String propertiesFile, ClassLoader classLoader) {
+    Properties properties = new Properties();
+    try {
+      LOGGER.debug(
+          "Attempting to load properties from {} with Spring PropertiesLoaderUtils with class loader.",
+          propertiesFile);
+      if (classLoader != null) {
+        properties = PropertiesLoaderUtils.loadAllProperties(propertiesFile, classLoader);
+      } else {
+        properties =
+            PropertiesLoaderUtils.loadAllProperties(
+                propertiesFile, PropertiesLoader.class.getClassLoader());
       }
-      properties = filtered;
+    } catch (IOException e) {
+      LOGGER.debug("Unable to load properties using default Spring properties loader.", e);
+    }
+    return properties;
+  }
 
+  /**
+   * Try loading the properties directly from the file system. If the properties file has a
+   * fully-qualified absolute path (which is what the blueprint file should specify) then it can be
+   * loaded directly, using this method. Otherwise the path will be considered relative and attempts
+   * will be made with {@code karaf.home} and {@code ddf.home} property values prepended to the
+   * original path.
+   */
+  @SuppressWarnings("squid:S1172" /* Used in bi-function */)
+  @VisibleForTesting
+  static Properties attemptLoadWithFileSystem(String propertiesFile, ClassLoader classLoader) {
+    LOGGER.debug("Attempting to load properties from file system: {}", propertiesFile);
+    Properties properties = new Properties();
+
+    String karafHome = System.getProperty("karaf.home");
+    String ddfHome = System.getProperty("ddf.home");
+
+    File propFile;
+    AbsolutePathResolver absPath = new AbsolutePathResolver(propertiesFile);
+    if (StringUtils.isNotBlank(karafHome)) {
+      propFile = new File(absPath.getPath(karafHome));
+    } else if (StringUtils.isNotBlank(ddfHome)) {
+      propFile = new File(absPath.getPath(ddfHome));
     } else {
-      LOGGER.debug("Properties file must not be null.");
+      propFile = new File(propertiesFile);
+    }
+
+    if (propFile.exists()) {
+      try (InputStreamReader reader =
+          new InputStreamReader(new FileInputStream(propFile), StandardCharsets.UTF_8)) {
+        properties.load(reader);
+      } catch (FileNotFoundException e) {
+        LOGGER.debug("Could not find properties file: {}", propFile.getAbsolutePath(), e);
+      } catch (IOException e) {
+        LOGGER.debug("Error reading properties file: {}", propFile.getAbsolutePath(), e);
+        properties.clear();
```

Comment:
❓ Is this what we want to do if we fail to close? Clear out the read properties? I can't see why we wouldn't just log the failed close, if that's what happened, and then return the read contents.

Reply:
The goal of this ticket is to refactor and preserve functionality, whether that functionality be good or bad. Changing functionality could result in not only our code but integrator's code breaking as well.